server/datastore/mysql/hosts.go

```diff
@@ -2266,6 +2266,9 @@ func (ds *Datastore) SetOrUpdateDeviceAuthToken(ctx context.Context, hostID uint
 `
 	_, err := ds.writer(ctx).ExecContext(ctx, stmt, hostID, authToken)
 	if err != nil {
+		if strings.Contains(err.Error(), "Duplicate entry 'token'") {
```

Comment:
Double checking: does `isDuplicate` work?

https://github.com/fleetdm/fleet/blob/eb9075026cb772b9c34415adc803ed9aef922d38/server/datastore/mysql/errors.go#L109-L117

Reply:
yes, `isDuplicate` works here. I updated the code.